orchagent/bufferorch.cpp

```diff
@@ -790,20 +857,36 @@ task_process_status BufferOrch::processQueue(KeyOpFieldsValuesTuple &tuple)
         for (size_t ind = range_low; ind <= range_high; ind++)
         {
             SWSS_LOG_DEBUG("processing queue:%zd", ind);
-            if (port.m_queue_ids.size() <= ind)
+            sai_object_id_t queue_id;
+
+            if (gMySwitchType == "voq") 
             {
-                SWSS_LOG_ERROR("Invalid queue index specified:%zd", ind);
-                return task_process_status::task_invalid_entry;
-            }
-            if (port.m_queue_lock[ind])
+                std :: vector<sai_object_id_t> queue_ids = gPortsOrch->getPortVoQIds(port);
```

Comment:
looks like it can be called only once out of the loop.

Reply:
Can you please elaborate ? If you are comparing this with regular queue, i am not using port.m_queue_lock[ind] check. Is that a problem ?